### ISSUE TITLE
Train the Avg Perceptron POS model till converged.

### ISSUE THREAD

**Issue by Ayushk4** [JuliaText/TextAnalysis.jl#188]
As of now, the Avg Perceptron POS model added in #131 gives ~60 per cent accuracy [on CoNLL 2003](https://github.com/Ayushk4/POS.jl/blob/master/valid/CoNLL.ipynb) which is decent for the 30+ classes it addresses. But still is low compared to claimed ~90% accuracy.

We can port weights from other libraries (depending on the licence permissions). Alternatively, we can train those.

But first, it will be better to know precisely the current accuracy Avg Perceptron POS Tagger offers on CoNLL 2000 and GMB as well (CorpusLoaders.jl provides the APIs for these two datasets.)

**Comment by tejasvaidhyadev** [comment]
Hi @Ayushk4 
CorpusLoaders.jl APIs can not be directly used because of  datatype constraint input require in Avg Perceptron POS model. I think we can provide new methods in Avg Perceptron POS model or CorpusLoaders.jl to make them compatibile with each other.
As i was working with Pretrain weight `predict` function is returning missing values.(Accuracy on test set by CorpusLoder.jl  is 0.47(using Pretain model )
While training model is giving some unexcepted error.

**Comment by tejasvaidhyadev** [comment]
Hi @Ayushk4  
The process is getting killed due to unknown reasons during training on the CoNLL dataset.
I am attaching [gist ](https://gist.github.com/tejasvaidhyadev/01ea565ee8807a59c672866e4057c360)file of code used (Sorry, code is little messy)
I will be updating the status soon

**Comment by Ayushk4** [comment]
Hi, sorry for the late response.

Can you upload the code as notebook format, or fix the indentation in gistfile to make the code more readable?

**Comment by Ayushk4** [comment]
It will be great if you could also share the notebook used for measuring the performance of Avg. Preceptoron tagger on CoNLL and specify which CoNLL dataset you used?

I personally like the idea of writing new APIs to handle different data types and documents that TextAnalysis.jl provides.
Feel free to open an issue (or better yet send a PR) for the same.

**Comment by tejasvaidhyadev** [comment]
[Updated gist](https://gist.github.com/tejasvaidhyadev/01ea565ee8807a59c672866e4057c360) with comments and I will soon upload the Notebook used for measuring the performance of Avg. Preceptoron tagger on CoNLL `Test`(set)
Should I add API in `TextAnalysis.jl` or `CorpusLoaders.jl`?

**Comment by Ayushk4** [comment]
For CoNLL and dataset related APIs, they go to CorpusLoaders.jl.

For perception tagger related APIs supporting new inputs, they go to TextAnalysis.jl.
